SPECS/kubernetes/kubernetes-1.17.spec

```diff
@@ -0,0 +1,207 @@
+
+%prep
+%setup -q -D -T -b 0 -n %{name}
+
+# note: kubernetes RPM can be build from binaries provided in source0 tarball
+#       by doing nothing in %build and %check sections
+%build
+# expand kubernetes source tarball (which is included source0 tarball)
+echo "+++ extract sources from tarball"
+mkdir -p %{_builddir}/%{name}/src
+cd %{_builddir}/%{name}/src
+tar -xof %{_builddir}/%{name}/kubernetes-src.tar.gz
+
+# build and update kubernetes components that are provided as binary
+# (other/unused kubernetes componenents will not be built)
+components_to_build=$(ls -1 %{_builddir}/%{name}/node/bin)
+for component in ${components_to_build}; do
+  echo "+++ building ${component}"
+  make WHAT=cmd/${component}
+  cp -f _output/local/bin/linux/%{archname}/${component} %{_builddir}/%{name}/node/bin
+done
+
+%check
+cd %{_builddir}/%{name}/src
+components_to_test=$(ls -1 %{_builddir}/%{name}/node/bin)
+
+# perform unit tests
+# Note:
+#   - components are not unit tested the same way
+#   - not all components have unit
+for component in ${components_to_test}; do
+  if [[ ${component} == "kubelet" || ${component} == "kubectl" ]]; then
+    echo "+++ unit test pkg ${component}"
+    make test WHAT=./pkg/${component}
+  elif [[ ${component} == "kube-proxy" ]]; then
+    echo "+++ unit test pkg ${component}"
+    make test WHAT=./pkg/proxy
+  else
+    echo "+++ no unit test available for ${component}"
+  fi
+done
+
+%install
+# install binaries
+install -m 755 -d %{buildroot}%{_bindir}
+cd %{_builddir}
+binaries=(kubelet kubectl kubeadm)
+for bin in "${binaries[@]}"; do
+  echo "+++ INSTALLING ${bin}"
+  install -p -m 755 -t %{buildroot}%{_bindir} %{name}/node/bin/${bin}
+done
+
+# install service files
+install -d -m 0755 %{buildroot}/%{_lib}/systemd/system
+install -p -m 644 -t %{buildroot}%{_lib}/systemd/system %{SOURCE1}
+
+# install config files
+install -d -m 0755 %{buildroot}%{_sysconfdir}/kubernetes
+install -d -m 644 %{buildroot}%{_sysconfdir}/kubernetes/manifests
+
+# install the place the kubelet defaults to put volumes
+install -dm755 %{buildroot}%{_sharedstatedir}/kubelet
+install -dm755 %{buildroot}%{_var}/run/kubernetes
+
+install -d -m 0755 %{buildroot}/%{_lib}/tmpfiles.d
+cat << EOF >> %{buildroot}/%{_lib}/tmpfiles.d/kubernetes.conf
+d %{_var}/run/kubernetes 0755 kube kube -
+EOF
+
+%clean
+rm -rf %{buildroot}/*
+
+%pre
+if [ $1 -eq 1 ]; then
+    # Initial installation.
+    getent group kube >/dev/null || groupadd -r kube
+    getent passwd kube >/dev/null || useradd -r -g kube -d / -s /sbin/nologin \
+            -c "Kubernetes user" kube
+fi
+
+%post
+chown -R kube:kube %{_sharedstatedir}/kubelet
+chown -R kube:kube %{_var}/run/kubernetes
+systemctl daemon-reload
+
+%post kubeadm
+systemctl daemon-reload
+systemctl stop kubelet
+systemctl enable kubelet
+
+%postun
+if [ $1 -eq 0 ]; then
+    # Package deletion
+    userdel kube
+    groupdel kube
+    systemctl daemon-reload
+fi
+
+%files
+%defattr(-,root,root)
+%license LICENSES
```

Comment:
Where is LICENCES coming from?

Reply:
the LICENCES file come from kubernetes-node-linux-amd64.tar.gz tarball. By default %license macro use file from the root folder where source0 tarball has been expended.
and yes it took me a while to realize that ;)